src/en/sunshinebutterflyscans/src/eu/kanade/tachiyomi/extension/en/sunshinebutterflyscans/Dto.kt

```diff
@@ -0,0 +1,86 @@
+package eu.kanade.tachiyomi.extension.en.sunshinebutterflyscans
+
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+class Dto(
+    val series: String,
+    val timestamp: String,
+    val num: Int,
+    @SerialName("chname") val chapterName: String,
+    @SerialName("AlbumID") val albumID: String,
+    @SerialName("projectname") val projectName: String,
+    @SerialName("projectdesc") val projectDesc: String,
+    @SerialName("projectaltname") val altName: String,
+    @SerialName("projectauthor") val projectAuthor: String,
+    @SerialName("projectartist") val projectArtist: String,
+    @SerialName("projectthumb") val projectThumb: String,
+    @SerialName("projectstatus") val projectStatus: String,
+    @SerialName("projecttags") val projectTags: String,
+) {
+    fun toSManga(cdnUrl: String): SManga = SManga.create().apply {
+        title = series
+        thumbnail_url = cdnUrl + projectThumb
+        url = "/projects?n=$projectName"
+        description = buildString {
+            append(projectDesc.nonEmpty())
+            if (altName.isNotEmpty()) {
+                append("\n\n")
+                append("Alternative name: ")
+                append(altName)
+            }
+        }
+        genre = projectTags.nonEmpty()?.replace(",", ", ")
+        status = projectStatus.toStatus()
+        author = projectAuthor.nonEmpty()
+        artist = projectArtist.nonEmpty()
+        initialized = true
+    }
+
+    private fun String.toStatus(): Int = when (this) {
+        "current" -> SManga.ONGOING
+        "complete" -> SManga.COMPLETED
+        "dropped" -> SManga.CANCELLED
+        "licensed" -> SManga.LICENSED
```

Comment:
the app doesn't show chapters when status is `Licensed`. so if there are chapters for such manga on the site then consider not setting this status

Reply:
The site doesn't display them